Source/Csla/Core/ContextDictionary.cs

```diff
@@ -104,19 +113,14 @@ public void Remove(object key)
     /// <inheritdoc cref="System.Collections.ICollection.SyncRoot"/>
     public object SyncRoot
     {
-      get => ((IDictionary)this).SyncRoot;
+      get => throw new NotSupportedException("SyncRoot NotSupported");
```

Comment:
English text can't be hard-coded. Either the text needs to come from the Resources file or avoid using text at all.

In this case just `throw new NotSupportedException(nameof(SyncRoot));`

Reply:
Changed to use resource